src/taxonomy/NcbiTaxonomy.cpp

```diff
@@ -104,89 +114,45 @@ std::vector<std::string> splitByDelimiter(const std::string &s, const std::strin
     return result;
 }
 
-void NcbiTaxonomy::loadNodes(const std::string &nodesFile) {
+size_t NcbiTaxonomy::loadNodes(const std::string &nodesFile) {
+    Debug(Debug::INFO) << "Loading nodes file ...";
     std::ifstream ss(nodesFile);
     if (ss.fail()) {
         Debug(Debug::ERROR) << "File " << nodesFile << " not found!\n";
         EXIT(EXIT_FAILURE);
     }
 
+    std::map<TaxID, int> Dm; // temporary map TaxID -> internal ID;
+    int maxTaxID = 0;
+    int currentId = 0;
     std::string line;
     while (std::getline(ss, line)) {
         std::vector<std::string> result = splitByDelimiter(line, "\t|\t", 3);
-        int currentId = (int) strtol(result[0].c_str(), NULL, 10);
-        int parentId = (int) strtol(result[1].c_str(), NULL, 10);
-
-        // no circular references
-        if (currentId == parentId) {
-            continue;
-        }
-
-        std::map<int, TaxonNode>::iterator it = taxonTree.find(currentId);
-        if (it != taxonTree.end()) {
-            (*it).second.parentTaxon = parentId;
-            (*it).second.rank = result[2];
-        } else {
-            taxonTree.emplace(currentId, TaxonNode(currentId, parentId, result[2]));
-        }
-
-        it = taxonTree.find(parentId);
-        if (it != taxonTree.end()) {
-            (*it).second.children.emplace_back(currentId);
-        } else {
-            TaxonNode node(parentId, 0, "");
-            node.children.emplace_back(currentId);
-            taxonTree.emplace(parentId, node);
+        TaxID taxId = (TaxID) strtol(result[0].c_str(), NULL, 10);
+        TaxID parentTaxId = (TaxID) strtol(result[1].c_str(), NULL, 10);
+        if (taxId > maxTaxID) {
+            maxTaxID = taxId;
         }
+        taxonNodes.emplace_back(TaxonNode(currentId, taxId, parentTaxId, result[2]));
```

Comment:
```suggestion
        taxonNodes.emplace_back(currentId, taxId, parentTaxId, result[2]);
```
Let it automatically forward the parameters with emplace_back